tasks/quality_gates.py

```diff
@@ -0,0 +1,158 @@
+    body_info = body_info_pattern
+    body_error = body_error_pattern
+    body_error_feet = body_error_feet_pattern
+
+    withError = False
+    withInfo = False
+    # Sort gates by error_types to group in between NoError, AssertionError and StackTrace
+    for gate in sorted(gateStates, key=lambda x: x["error_type"] is None):
+
+        def getMetric(metric_name, gate_name=gate['name']):
+            try:
+                return metricHandler.get_formated_metric(gate_name, metric_name)
+            except KeyError:
+                return "DataNotFound"
+
+        if gate["error_type"] is None:
+            body_info += f"|{SUCCESS_CHAR}|{gate['name']}|{getMetric('current_on_disk_size')}|{getMetric('max_on_disk_size')}|{getMetric('current_on_wire_size')}|{getMetric('max_on_wire_size')}|\n"
+            withInfo = True
+        else:
+            body_error += f"|{FAIL_CHAR}|{gate['name']}|{getMetric('current_on_disk_size')}|{getMetric('max_on_disk_size')}|{getMetric('current_on_wire_size')}|{getMetric('max_on_wire_size')}|\n"
+            body_error_feet += f"|{gate['name']}|{gate['error_type']}|{format_gate_message(gate['message'])}|\n"
+            withError = True
+
+    body_error_feet += "\n</details>\n"
+    body = f"Please find below the results from static quality gates\n{body_error+body_error_feet if withError else ""}\n\n{body_info if withInfo else ""}"
+
+    pr_commenter(ctx, title=title, body=body)
+
+
+def _print_quality_gates_report(gateStates: list[dict[str, typing.Any]]):
+    print(color_message("======== Static Quality Gates Report ========", "magenta"))
+    for gate in sorted(gateStates, key=lambda x: x["error_type"] is not None):
+        if gate["error_type"] is None:
+            print(color_message(f"Gate {gate['name']} succeeded {SUCCESS_CHAR}", "blue"))
+        elif gate["error_type"] == "AssertionError":
+            print(
+                color_message(
+                    f"Gate {gate['name']} failed {FAIL_CHAR} because of the following assertion failures :\n{gate['message']}",
+                    "orange",
+                )
+            )
+        else:
+            print(
+                color_message(
+                    f"Gate {gate['name']} failed {FAIL_CHAR} with the following stack trace :\n{gate['message']}",
+                    "orange",
+                )
+            )
+
+
+@task
+def parse_and_trigger_gates(ctx, config_path="test/static/static_quality_gates.yml"):
+    """
+    Parse and executes static quality gates
+    :param ctx: Invoke context
+    :param config_path: Static quality gates configuration file path
+    :return:
+    """
+    with open(config_path) as file:
+        config = yaml.safe_load(file)
+
+    gateList = list(config.keys())
+    quality_gates_mod = __import__("tasks.static_quality_gates", fromlist=gateList)
+    print(f"{config_path} correctly parsed !")
+    metricHandler = GateMetricHandler(
```

Comment:
```suggestion
    metric_handler = GateMetricHandler(
```
There are a lot of them I let you handle the naming everywhere, thanks in advance